flaml/model.py

```diff
@@ -575,28 +603,23 @@ def on_epoch_end(self, args, state, control, **callback_kwargs):
             callbacks=[EarlyStoppingCallbackForAuto],
         )
 
-        setattr(self._trainer, "_use_ray", self.use_ray)
         if self._task in NLG_TASKS:
             setattr(self._trainer, "_is_seq2seq", True)
         if kwargs.get("gpu_per_trial"):
             self._trainer.args._n_gpu = kwargs.get("gpu_per_trial")
+
+        if kwargs.get("gpu_per_trial") == 0:
+            os.environ["CUDA_VISIBLE_DEVICES"] = ""

```

Comment:
1. also handle the case where gpu_per_trial>0. Note that if ray already restricts the visible devices to the right number, no modification should be done.
2. recover it after training.